src/main/java/org/traccar/protocol/LaipacProtocolEncoder.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2019 Anton Tananaev (anton@traccar.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.protocol;
+
+import org.traccar.Protocol;
+import org.traccar.StringProtocolEncoder;
+import org.traccar.model.Command;
+import org.traccar.helper.Checksum;
+
+public class LaipacProtocolEncoder extends StringProtocolEncoder {
+
+    private final Protocol protocol;
+    private final String defaultDevicePassword;
```

Comment:
Same question here. Also, it should probably be defined as a constant in a single place.

Reply:
Do you think it can be a public member of LaipacProtocol class? It needs to be usable from both the encoder and the decoder. I don't want it to be a constant because I'm planning to make it configurable (if you like the idea, using a server wide attribute)

Reply:
For now, would you like it to be a constant in LaipacProtocol accesible by a method?

Reply:
No, usually constants go to decoder.

Reply:
So... What about the first idea?

Reply:
Default should not be configurable. We discussed it.

Reply:
So, just to know if I understand. You are saying that we should not have a configurable default password for every device of a type (using the same protocol), and we should configure the password for each device, even if it is the same non hard-coded default?

Reply:
Default is default. It should be a constant. Password should be configurable.

Reply:
Sorry, now I remember you said to keep it static for this PR. But, what do you suggest not being a private static constant in LaipacProtocol, accessible through a method from both the encoder and the decoder?

Reply:
Make it a static constant in decoder, like all other constants there.

Reply:
And what about the code in the encoder? It needs the password too to send commands to the device.

Reply:
What's the problem with accessing constant from encoder.

Reply:
No problem at all, but we need to declare the same constant in both places. Is that ok for you?

Reply:
Please define it in a single place, as I mentioned previously.

Reply:
Sorry bothering you. That's my point... What single place (common to both encoder and decoder) do you suggest not being LaipacProtocol class? Thanks...

Reply:
I said it twice - decoder!

Reply:
Sorry again. Do you mean to declare the constant as a public static constant in the decoder class?

Reply:
Yes, please check how it's done in other decoders.

Reply:
In general follow existing practice. Don't try to invent anything.

Reply:
Oh, I see your point now. I'll follow your guideline. I'll be doing some tests and come with more commits. Thank you for your patience.